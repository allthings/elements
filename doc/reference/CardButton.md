<!-- 
This is an auto-generated markdown. 
You can change it in "/Users/daniel/Dev/allthings/elements/src/CardButton/CardButton.tsx" and run build:docs to update this file.
-->
# CardButton
CardButton can to enable users to do actions directly related to content on
on cards. It should always go into a [CardFooter](CardFooter.md).
## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
|backgroundColor|string|
|autoFocus|boolean|
|htmlElement|string|
|alignH|"none", "start", "center", "end", "space-around", "space-between"|horizontal alignment
|alignV|"none", "start", "center", "end", "stretch"|vertical alignment
|direction|"row", "column", "row-reverse", "column-reverse"|direction
|fill|boolean|Passing true, will make the view fill out available space
|wrap|"wrap", "inherit", "initial", "nowrap", "wrap-reverse"|Defining how children will wrap
|flex|number, "flex", "none", "initial", "nogrow", "grow", "auto", "noshrink"|Flex values, can be 5, 10, 15 ... 100 or 33, 66
