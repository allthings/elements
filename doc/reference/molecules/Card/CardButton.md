# CardButton

## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
|children|node|
|onClick|func|<br>Default: noop => noop
