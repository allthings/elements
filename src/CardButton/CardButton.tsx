import React, { FunctionComponent } from 'react'
import View, { IViewProps } from '../View'
import { css } from 'glamor'
import { color as col, lightness } from 'kewler'
import Theme from '../Theme'
import { color } from '../utils/propTypes/color'

const style = (backgroundColor: string) =>
  css({
    backgroundColor,
    border: 'none',
    transition: '250ms ease-in-out',
    ':focus': {
      outline: 'none',
    },
    ':hover': {
      cursor: 'pointer',
      background:
        backgroundColor.indexOf('#') !== -1
          ? col(backgroundColor, lightness(-10))
          : backgroundColor,
    },
  })

interface ICardButtonProps extends IViewProps {
  readonly backgroundColor?: color
  readonly autoFocus?: boolean
}

/**
 * CardButton can to enable users to do actions directly related to content on
 * on cards. It should always go into a [CardFooter](CardFooter.md).
 */
const CardButton: FunctionComponent<ICardButtonProps> = ({
  backgroundColor = '#ffffff',
  ...props
}) => (
  <Theme>
    {({ colorize }) => (
      <View
        alignH="center"
        flex="flex"
        alignV="center"
        direction="row"
        htmlElement="button"
        {...style(colorize(backgroundColor))}
        {...props}
      />
    )}
  </Theme>
)

export default CardButton
