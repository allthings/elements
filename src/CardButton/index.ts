export { default } from './CardButton'
export * from './CardButton'
