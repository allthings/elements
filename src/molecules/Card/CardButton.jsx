import PropTypes from 'prop-types'
import React from 'react'
import View from '../../atoms/View'
import { css } from 'glamor'

const style = css({
  ':hover': {
    cursor: 'pointer',
    background: '#f9f9f9',
  },
})

export default function CardButton({
  children,
  onClick = noop => noop,
  ...props
}) {
  return (
    <View
      alignH="center"
      flex="flex"
      alignV="center"
      direction="row"
      onClick={onClick}
      {...style}
      {...props}
    >
      {children}
    </View>
  )
}

CardButton.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
}
