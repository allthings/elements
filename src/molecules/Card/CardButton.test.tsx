import React from 'react'
import renderer from 'react-test-renderer'
import CardButton from './CardButton'
import { ThemeProvider } from '../..'

test('CardButton renders without error', () => {
  const tree = renderer
    .create(
      <ThemeProvider>
        <CardButton onClick={_ => _} backgroundColor="primary">
          Click me
        </CardButton>
      </ThemeProvider>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('CardButton renders custom color', () => {
  const tree = renderer
    .create(<CardButton backgroundColor="#f0f0f0">Click me</CardButton>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
