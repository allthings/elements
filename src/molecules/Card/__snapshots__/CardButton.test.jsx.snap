// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardButton renders custom color 1`] = `
.css-yd6zvg,
[data-css-yd6zvg] {
  box-sizing: border-box;
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: center;
  cursor: pointer;
  flex: 1;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-flex: 1;
}

.css-keghjr,
[data-css-keghjr] {
  background-color: #f0f0f0;
  transition: 500ms ease-in-out;
  -webkit-transition: 500ms ease-in-out;
  -moz-transition: 500ms ease-in-out;
}

.css-keghjr:hover,
[data-css-keghjr]:hover {
  cursor: pointer;
  background: #d9d9d9;
}

<div
  data-css-keghjr=""
  data-css-yd6zvg=""
  onClick={[Function]}
>
  Click me
</div>
`;

exports[`CardButton renders without error 1`] = `
.css-7wo8x0,
[data-css-7wo8x0] {
  background-color: #3598db;
  transition: 500ms ease-in-out;
  -webkit-transition: 500ms ease-in-out;
  -moz-transition: 500ms ease-in-out;
}

.css-7wo8x0:hover,
[data-css-7wo8x0]:hover {
  cursor: pointer;
  background: #227dbf;
}

.css-yd6zvg,
[data-css-yd6zvg] {
  box-sizing: border-box;
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: center;
  cursor: pointer;
  flex: 1;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-flex: 1;
}

<div
  data-css-7wo8x0=""
  data-css-yd6zvg=""
  onClick={[Function]}
>
  Click me
</div>
`;
